File: observer/python/classes/sms_subscriber.py
from interfaces.observer import Observer

class SMSSubscriber(Observer):
    def __init__(self, phone_number):
        self.phone_number = phone_number

    def update(self, news):
        """Notificación por SMS"""
        print(f"Enviando SMS al {self.phone_number}: Noticias - {news}")

File: observer/java/classes/SMSSubscriber.java
package observer.java.classes;

import observer.java.interfaces.Observer;

public class SMSSubscriber implements Observer {
    private String phoneNumber;

    public SMSSubscriber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    @Override
    public void update(String news) {
        System.out.println("Enviando SMS al " + phoneNumber + ": Noticias - " + news);
    }
}
